yts/test_suite_test.go

```diff
@@ -123,110 +123,135 @@ func runTest(t *testing.T, testPath string) {
 
 		if expectError {
 			if unmarshalErr == nil {
-				t.Errorf("Test: %s\nDescription: %s\nError: Expected unmarshal error but got none", testPath, testDescription)
+				t.Errorf(
+					"Test: %s\nDescription: %s\nError: Expected unmarshal error but got none",
+					testPath, testDescription)
 			}
 			return
 		}
 		if unmarshalErr != nil {
-			t.Errorf("Test: %s\nDescription: %s\nError: Unexpected unmarshal error: %v", testPath, testDescription, unmarshalErr)
+			t.Errorf(
+				"Test: %s\nDescription: %s\nError: Unexpected unmarshal error: %v",
+				testPath, testDescription, unmarshalErr)
 		}
 	})
 
 	t.Run("EventComparisonTest", func(t *testing.T) {
 		shouldSkipTest(t)
-		expectedEventsPath := filepath.Join(testPath, "test.event")
-		if _, err := os.Stat(expectedEventsPath); err != nil {
-			return
-		}
-		expectedEventsBytes, err := os.ReadFile(expectedEventsPath)
-		if err != nil {
-			t.Errorf("Test: %s\nDescription: %s\nError: Failed to read test.event: %v", testPath, testDescription, err)
-			return
-		}
-		expectedEvents := normalizeLineEndings(string(expectedEventsBytes))
 		expectedEvents = strings.TrimSuffix(expectedEvents, "\n")
 		actualEvents, eventErr := getEvents(inYAML)
 
 		if expectError {
 			if eventErr == nil {
-				t.Errorf("Test: %s\nDescription: %s\nError: Expected error on event parsing but got none", testPath, testDescription)
+				t.Errorf(
+					"Test: %s\nDescription: %s\nError: Expected error on event parsing but got none",
+					testPath, testDescription)
 			}
 			return
 		}
 		if eventErr != nil {
-			t.Errorf("Test: %s\nDescription: %s\nError: Unexpected error on event parsing: %v", testPath, testDescription, eventErr)
+			t.Errorf("Test: %s\nDescription: %s\nError: Unexpected error on event parsing: %v",
+				testPath, testDescription, eventErr)
 			return
 		}
 		actualEventsStr := normalizeLineEndings(actualEvents)
 		if actualEventsStr != expectedEvents {
-			t.Errorf("Test: %s\nDescription: %s\nError: Event mismatch\nExpected:\n%q\nGot:\n%q", testPath, testDescription, expectedEvents, actualEventsStr)
+			t.Errorf(
+				"Test: %s\nDescription: %s\nError: Event mismatch\nExpected:\n%q\nGot:\n%q",
+				testPath, testDescription, expectedEvents, actualEventsStr)
 		}
 	})
 
-	// Only proceed with marshal and JSON tests if unmarshal was successful and no expected error
-
-	t.Run("MarshalTest", func(t *testing.T) {
+	t.Run("MarshalCanonicalTest", func(t *testing.T) {
 		shouldSkipTest(t)
-		var currentUnmarshaledValue any
-
-		currentUnmarshalErr := yaml.Unmarshal(inYAML, &currentUnmarshaledValue)
+		var unmarshaledValue interface{}
 
-		if !(currentUnmarshalErr == nil || expectError) {
+		if expectError || !fileExists(testPath, "out.yaml") {
```

Comment:
check for `expectError` should be done after `yaml.Unmarshal`

Reply:
I disagree. Maybe I got the wrong understanding (the test files didn't come with an instruction manual) but when an error is expected, the only test I think we should run is the `UnmarshallTest`.

Reply:
For the most part, those changes are fixes to what I think I got wrong when I originally wrote the tests.